src/lifecycles/load.js

```diff
@@ -18,9 +20,11 @@ export function toLoadPromise(app) {
       const loadPromise = app.loadImpl(getProps(app));
       if (!smellsLikeAPromise(loadPromise)) {
         // The name of the app will be prepended to this error message inside of the handleAppError function
-        throw Error(`single-spa loading function did not return a promise. Check the second argument to registerApplication('${app.name}', loadingFunction, activityFunction)`);
+        throw new UserError(`single-spa loading function did not return a promise. Check the second argument to registerApplication('${app.name}', loadingFunction, activityFunction)`);
       }
       return loadPromise.then(val => {
+        app.loadErrorHref = null;
```

Comment:
If the load succeeds, always reset `loadErrorHref` to null. Even if the load itself isn't valid, which will be handled below and the app will go into `SKIP_BECAUSE_BROKEN`, the point of the `loadErrorHref` is to only be tied to the actual `app.status === LOAD_ERROR` and if we've made it this far, it can't be in that state.

Reply:
👍